### PR TITLE
Adds plant hostility

### PR DESCRIPTION
**Pull request by MadmanMartian**
Ever wanted a plant to just slap intruders? Or fire lube-based products at people?

This will do just that.

Biggest problem currently is that a lot of an items attack code relies on the attacker being a mob, not an object, so this will touch a lot of that code and it will likely not be pretty.

:cl:
 * rscadd: Adds plant hostility traits. Level 1 it will attack people in proximity using its produce. Level 2 it will fire the produce at mobs in the same area as itself.

### CODE REVIEW

File: code/modules/hydroponics/hydro_tray.dm [hunk 1]
```diff
@@ -531,22 +531,33 @@
...
+							M.gib(meat = 0) //"meat" argument only exists for mob/living/simple_animal/gib()
+							nutrilevel += 6
+							check_level_sanity()
+							update_icon()
+
+		if(harvest && seed.hostile == 1 && !(isnull(seed.products) || !seed.products.len || seed.yield <= 0) && !M.isDead() && M.m_intent == "run") //We're angry, and something is nearby.
+			visible_message("<span class='warning'>\The [seed.display_name] lashes out at \the [M]!</span>")
+			playsound(src, "sound/weapons/whip_crack.ogg", 30, 1)
+			var/obj/item/I = seed.generate_product()
+			I.preattack(M, src, 1)
```
Comment (MadmanMartian) [inline code/modules/hydroponics/hydro_tray.dm:556]: I hate everything about this, and may just make an attack-generic function

**Comment by spoonylove** [comment]
plants vs assistants by popcap.

File: code/modules/hydroponics/hydro_tools.dm [hunk 2]
```diff
@@ -201,6 +201,12 @@
 		if(2)
 			dat	+= "<br>Its fruit is excessively soft and juicy."
 
+	switch(grown_seed.hostile)
+		if(1)
```
Comment (ShiftyRail) [inline code/modules/hydroponics/hydro_tools.dm:205]: Muh defines

File: code/modules/hydroponics/hydro_tray.dm [hunk 3]
```diff
@@ -531,22 +531,33 @@
...
-				spawn(30)
-					if(M && M.loc == get_turf(src))
-						unlock_atom(M)
-						M.gib(meat = 0) //"meat" argument only exists for mob/living/simple_animal/gib()
-						nutrilevel += 6
-						check_level_sanity()
-						update_icon()
+	if(seed && !dead)
+		if(seed.carnivorous == 2 && age > seed.maturation)
+			if(istype(M, /mob/living/simple_animal/mouse) || istype(M, /mob/living/simple_animal/hostile/lizard) && !M.locked_to && !M.anchored)
```
Comment (ShiftyRail) [inline code/modules/hydroponics/hydro_tray.dm:536]: Maybe make it a generic list of types I don't know

Comment (MadmanMartian) [reply code/modules/hydroponics/hydro_tray.dm:536]: Not my code, but okay

**Pull request closed without merging**